test/e2e/e2e.go

```diff
@@ -373,11 +374,74 @@ var _ = ginkgo.Describe("[efs-csi] EFS CSI", func() {
 			encryptInTransit := false
 			testEncryptInTransit(f, &encryptInTransit)
 		})
+
+		testReadWriteOncePod := func(f *framework.Framework, pvc *v1.PersistentVolumeClaim) {
+
+			err := e2epv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, f.ClientSet, f.Namespace.Name, pvc.Name,
+				time.Second*5, time.Minute)
+			framework.ExpectNoError(err, "waiting for pv to be provisioned and bound")
+
+			podSpec := e2epod.MakePod(f.Namespace.Name, nil, []*v1.PersistentVolumeClaim{pvc}, false, "")
+			podSpec.Spec.RestartPolicy = v1.RestartPolicyNever
+			pod1, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(context.TODO(), podSpec, metav1.CreateOptions{})
+			framework.ExpectNoError(err, "creating pod")
+
+			err = e2epod.WaitForPodRunningInNamespace(f.ClientSet, pod1)
+			framework.ExpectNoError(err, "pod started running successfully")
+
+			pod2, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(context.TODO(), podSpec, metav1.CreateOptions{})
+			err = e2epod.WaitForPodNotPending(f.ClientSet, f.Namespace.Name, pod2.Name)
+			framework.ExpectError(err, "second pod could not mount PVC as well")
```

Comment:
`pod2` in this case will fail during scheduling because we see `pod1` is already using the PVC. If you want to test enforcement on a single node you can try setting `Pod.Spec.NodeName`.

Reply:
Hi Chris, thanks for the comments. Isn't this what we want to test? Or are you saying we should be more specific and allow the pod to be allocated to a node to then make sure it fails because of the PVC and not other conditions?

Reply:
I'm saying there's two ways ReadWriteOncePod can be enforced, 1) is during scheduling, and 2) is during mounting. The only way to test 2) is to force two pods on the same node by setting the node name.

A separate heads up, I'm working on implementing a similar e2e test in upstream k8s (I have to add it now so we can collect signal for 1.26 before beta graduation in 1.27). I'll tag you on the PR when it's available.

Reply:
Oh of course, I'll see if I can cook something up that does that, it feels like an important test. If you want to just cherry pick what's here feel free